parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java

```diff
@@ -108,32 +99,46 @@ private static long end(List<BlockMetaData> blocks) {
   }
 
   /**
+   * Builds a {@code ParquetInputSplit} from a mapreduce {@link FileSplit}.
+   *
+   * @param split a mapreduce FileSplit
+   * @return a ParquetInputSplit
+   * @throws IOException
+   */
+  public static ParquetInputSplit from(FileSplit split) throws IOException {
+    return new ParquetInputSplit(split.getPath(),
+        split.getStart(), split.getStart() + split.getLength(),
+        split.getLength(), split.getLocations(), null);
+  }
+
+  /**
+   * Builds a {@code ParquetInputSplit} from a mapred
+   * {@link org.apache.hadoop.mapred.FileSplit}.
+   *
+   * @param split a mapreduce FileSplit
+   * @return a ParquetInputSplit
+   * @throws IOException
+   */
+  public static ParquetInputSplit from(org.apache.hadoop.mapred.FileSplit split) throws IOException {
+    return new ParquetInputSplit(split.getPath(),
```

Comment:
Actually, I took another look at this. I think the right thing to do is to update Hive so that it doesn't need an internal Parquet split class at all. It should pass `FileSplit` to Parquet instead now that Parquet's splits are basically file splits.